packages/checkbox/src/react/index.tsx

```diff
@@ -70,108 +69,93 @@ const Checkbox = React.forwardRef<HTMLInputElement, CheckboxProps>(
     React.useEffect(
       function updateIndetermiateAttr() {
         if (!ref.current) return
-        ref.current.indeterminate = props.indeterminate || false
+        ref.current.indeterminate = indeterminate || false
       },
-      [props.indeterminate]
+      [indeterminate]
     )
 
-    const handleKeyDown = (evt: React.KeyboardEvent<HTMLLabelElement>) => {
-      if (evt.key === 'Enter' || evt.key === ' ') handleClick(evt)
-    }
-
-    const handleClick = (
+    const toggle = (
       evt:
-        | React.KeyboardEvent<HTMLLabelElement>
         | React.MouseEvent<HTMLLabelElement>
+        | React.KeyboardEvent<HTMLLabelElement>
     ) => {
+      if (onCheck && isFunction(onCheck)) onCheck(evt, !checked, value, name)
+      if (square.current) square.current.focus()
+    }
+
+    const handleClick: React.MouseEventHandler<HTMLLabelElement> = evt => {
+      if (disabled) return
+
       evt.preventDefault()
       evt.stopPropagation()
+      evt.persist()
 
-      const { onCheck } = props
-      if (typeof onCheck === 'function') {
-        onCheck(evt, !props.checked, props.value, props.name)
-      }
+      toggle(evt)
+    }
 
-      if (square.current) square.current.focus()
+    const handleKeyDown: React.KeyboardEventHandler<HTMLLabelElement> = evt => {
+      if (disabled) return
+
+      evt.persist()
+      if (evt.key === 'Enter' || evt.key === ' ') toggle(evt)
```

Comment:
native checkboxes only response to space....we also support `Enter`...i just left it